src/nodes/ImageView.vue

```diff
@@ -295,6 +301,15 @@ export default {
 			tr.delete(pos, pos + this.node.nodeSize)
 			this.view.dispatch(tr)
 		},
+		async setImageSrc(imageUrl) {
+			const file = await axios.get(imageUrl)
+			const mime = file.headers['content-type']
+			if (mime === 'image/svg+xml') {
+				this.imgSrc = `data:${mime};base64,${btoa(file.data)}`
+			} else {
+				this.imgSrc = imageUrl
+			}
+		},
```

Comment:
This does not seem ideal since it needs to download the image to determine the content-type and then again to display it if it's not an svg. In case of large images this might increase traffic / slow things down.
I don't know if we can avoid it though. I see two options:
* using a `HEAD` request instead that only sends metadata like the mimetype
* using the image in the src as it used to be, responding with some specific error response if it's an svg and handling that response somehow. I have no idea if the browser supports handling failing image request responses.

Reply:
Okay... so looks like there is [onerror](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement#errors) handling for images but i don't see any way to inspect the cause of the error. So the only option would be to handle all errors the same way - which might be:
1. load image with src
2. server rejects it because its an svg
3. javascript catches error tries to initiate alternative image with data property by fetching data from other url
4. server sees other url request and hands out svg
5. browser renders the svg.

Downside being that the second request would also trigger for images not loading for other reasons.

Reply:
Oh... and we would need to make sure that the request in 4. cannot originate from an image src as people can modify that. Including a token that cannot be determined by attackers or a non GET request seems like an option. The first seems cleaner the latter seems easier.

Reply:
Thought about this some more. The best of both worlds to me seems to be to:
1. attempt to load the image via src
2. trigger a second request at the same time that fetches image metadata and where the server includes the data if it's an svg image (might be a `PROPFIND`).
3. have an onerror handler for 1. that will check if 2. returned an svg mimetype and some data. If it did fill the data into the `<img>` replacing the src property.

Since only 1. or 2. will return the actual data traffic should stay roughly the same.
Since both 1. and 2. get started early there should not be additional delays.
Since 2. returns metadata and a data blob in it it cannot be rendered as an image as is. So that prevents using its url in an image src to get the svg to load via the src property.